src/transformers/trainer_utils.py

```diff
@@ -311,6 +331,18 @@ class TrainerMemoryTracker:
     }
 
     def __init__(self, skip_memory_metrics=False):
+
+        self.skip_memory_metrics = skip_memory_metrics
+
+        if not is_psutil_available():
+            # soft dependency on psutil
+            self.skip_memory_metrics = True
```

Comment:
Maybe add a logger.info telling the user they can `pip isntall psutil` to get automatic memory metrics?

Reply:
Since it will happen on every run, if they don't want it, that could be annoying.

It's already in `examples/_tests_requirements.txt` so at least users running the examples will already have it, the rest will be documented.